src/browser/modules/Stream/SysInfoFrame/index.jsx

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component } from 'preact'
+import { connect } from 'preact-redux'
+import { withBus } from 'preact-suber'
+import { CYPHER_REQUEST } from 'shared/modules/cypher/cypherDuck'
+import { isACausalCluster } from 'shared/modules/features/featuresDuck'
+import FrameTemplate from 'browser/modules/Stream/FrameTemplate'
+import FrameError from 'browser/modules/Stream/FrameError'
+import { SysInfoTableContainer, SysInfoTable, SysInfoTableEntry } from 'browser-components/Tables'
+import { toHumanReadableBytes } from 'services/utils'
+import { mapSysInfoRecords, getTableDataFromRecords } from './sysinfo'
+import Render from 'browser-components/Render'
+
+export class SysInfoFrame extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      error: '',
+      results: false
+    }
+  }
+
+  clusterResponseHandler () {
+    return (res) => {
```

Comment:
Why return function rather than just handle the response directly?
It looks a bit unusual.